test/org/jetbrains/plugins/bsp/project/importing/BspResolverLogicProperties.scala

```diff
@@ -65,35 +63,42 @@ class BspResolverLogicProperties extends AssertionsForJUnit with Checkers {
     }
   )
 
-  @Test @Ignore
-  def `test createScalaModuleDescription`(): Unit = check(
-    forAll(genPath, Gen.listOf(genBuildTargetTag)) { (basePath: Path, tags: List[String]) =>
+  def `test createScalaModuleDescription`(tagGenerator: Gen[String],assertions: (BuildTarget, ModuleDescriptionData, List[String], Option[File],List[SourceDirectory], List[File], List[File]) => Prop): Unit = check(
+    forAll(genPath, Gen.listOf(tagGenerator)) { (basePath: Path, tags: List[String]) =>
       forAll(Gen.listOf(genSourceDirectoryUnder(basePath)), Gen.listOf(genSourceDirectoryUnder(basePath))) {
         (sourceRoots: List[SourceDirectory], resourceRoots: List[SourceDirectory]) =>
           forAll { (target: BuildTarget, moduleBase: Option[File], outputPath: Option[File], classpath: List[File], dependencySources: List[File], languageLevel: LanguageLevel) =>
             val description = createModuleDescriptionData(target, tags, moduleBase, outputPath, sourceRoots, resourceRoots, classpath, dependencySources, Some(languageLevel))
+            val verifyBasePath = (description.basePath == moduleBase) :| "base path should be set"
 
-            val p1 = (description.basePath == moduleBase) :| "base path should be set"
-            val p2 = (tags.contains(BuildTargetTag.LIBRARY) || tags.contains(BuildTargetTag
-              .APPLICATION)) ==>
-              (description.output == outputPath &&
-                description.targetDependencies == target.getDependencies.asScala &&
-                description.classpathSources == dependencySources &&
-                description.sourceDirs == sourceRoots &&
-                description.classpath == classpath) :|
-                s"data not correctly set for library or application tags. Result data was: $description"
-            val p3 = tags.contains(BuildTargetTag.TEST) ==>
-              (description.testOutput == outputPath &&
-                description.targetTestDependencies == target.getDependencies.asScala &&
-                description.testClasspathSources == dependencySources &&
-                description.testSourceDirs == sourceRoots &&
-                description.testClasspath == classpath) :|
-                s"data not correctly set for test tag. Result data was: $description"
-
-            p1 && p2 && p3
+            verifyBasePath && assertions(target, description, tags, outputPath, sourceRoots, classpath, dependencySources)
           }
       }
-    }
+    })
+
+  @Test
+  def `test createScalaModuleDescription for Libraries and Applications`(): Unit = `test createScalaModuleDescription`(genBuildTargetTagWithoutTest,
+    (target, description, tags, outputPath, sourceRoots, classpath, dependencySources) =>
+      ((tags.contains(BuildTargetTag.LIBRARY) || tags.contains(BuildTargetTag
+        .APPLICATION)) && !tags.contains(BuildTargetTag.TEST)) ==>
```

Comment:
Due to this condition, if the test tag is considered in the generator, most of the test cases are ignored and the test fails. Can you think of any other way of doing this, besides having a second generator without the test tag @jastice ?

Reply:
that seems reasonable, actually. Alternatively, parameterize the generator

Reply:
Do you think that's better? Since this generator lives inside the testkit itself